src/halmos/sevm.py

```diff
@@ -414,6 +415,8 @@ def push(self, v: Word) -> None:
             self.stack.append(simplify(v))
 
     def pop(self) -> Word:
+        if not self.stack:
+            raise StackUnderflowError()
```

Comment:
this used to fail with the raw Python `IndexError`, and caught by `pytest.raises(Exception)`

ruff rightfully points out that catching `Exception` is not great, so we can finally address the todo and wrap the exception in the semantically correct `StackUnderflowError`